libutils/libcrypto-compat.c

```diff
@@ -0,0 +1,409 @@
+/*
+ * Copyright 2016 The OpenSSL Project Authors. All Rights Reserved.
+ *

```

Comment:
@kacf what do you think about this C file and its license? As I note in the commit message, this file has been taken from the [OpenSSL wiki](https://wiki.openssl.org/index.php/OpenSSL_1.1.0_Changes#Downloads).

Reply:
Should be ok. The license below ensures that we can use it as per the OpenSSL license.